platforms/jvm/plugins-java/src/integTest/groovy/org/gradle/java/JavaCrossCompilationIntegrationTest.groovy

```diff
@@ -171,6 +187,51 @@ class JavaCrossCompilationIntegrationTest extends AbstractIntegrationSpec {
         file('build/docs/javadoc/Thing.html').text.matches("(?s).*Generated by javadoc \\(.*?\\Q${version}\\E.*")
 
         where:
-        version << javaVersionsToCrossCompileAgainst()
+        version << allJavaVersions()
+    }
+
+    def "can compile but cannot run tests against Java #version JDK"() {
+        def jvm = withJavaProjectUsingToolchainsForJavaVersion(version)
+
+        settingsFile << """
+            rootProject.name = 'oldjava'
+        """
+
+        buildFile << """
+            dependencies {
+                testImplementation("junit:junit:4.13")
+            }
+        """
+
+        file("src/test/java/ThingTest.java") << """
+            public class ThingTest {
+                @org.junit.Test
+                public void check() { }
+            }
+        """
+
+        executer.beforeExecute({
+            withArgument("-Porg.gradle.java.installations.paths=" + jvm.getJavaHome().getAbsolutePath())
+        })
```

Comment:
Should not need these lines. They are covered by `withJavaProjectUsingToolchainsForJavaVersion`